### PR TITLE
Add suppression ability

### LINKED ISSUE #61: Project references used for copying binaries to output are flagged as unneeded
In a project graph I have an instance where a test project uses `ProjectReference`s to projects whose outputs - typically executables - are needed in the output path for running tests. Is there a way to signal to ReferenceTrimmer that a specific reference should be ignored? Ignoring `RT0002` in the whole project hides other, real problems. This syntax does not work but was what came to mind to try:

`<ProjectReference Include="..\OtherProj\OtherProj.csproj" NoWarn="RT0002" />`

### PR DESCRIPTION
**Pull request by stan-sz**
Fixes #61

### CODE REVIEW

File: src/Tasks/CollectDeclaredReferencesTask.cs [hunk 1]
```diff
@@ -160,7 +160,7 @@ public override bool Execute()
 
                     foreach (string assemblyName in packageInfo.CompileTimeAssemblies)
                     {
-                        declaredReferences.Add(new DeclaredReference(assemblyName, DeclaredReferenceKind.PackageReference, packageReference.ItemSpec));
+                        declaredReferences.Add(new DeclaredReference(assemblyName, DeclaredReferenceKind.PackageReference, packageReference.ItemSpec, packageReference.GetMetadata("NoWarn")));
```
Comment (erikmav) [inline src/Tasks/CollectDeclaredReferencesTask.cs:163]: >"NoWarn" [](http://example.com/codeflow?start=179&length=8)

nit: Move `"NoWarn"` to constant

**Comment by erikmav** [comment]
Needs reference documentation in README.md in the `## How to use` section - perhaps:

### Disabling a rule on a reference
To turn off a rule on a specific project or package reference, add the relevant RTxxxx code to a `NoWarn` metadata attribute. For example:

```xml
<ProjectReference Include="../Other/Project.csproj" NoWarn="RT0002" />
```

**Comment by erikmav** [comment]
I would also move the `## Rules` section above `## How to use` to introduce the RT codes earlier.

---
In reply to: [1720026036](https://github.com/dfederm/ReferenceTrimmer/pull/72#issuecomment-1720026036) [](http://example.com/codeflow?ancestors=1720026036)

**Review by erikmav — APPROVED** [review]
:shipit:

**Comment by stan-sz** [comment]
I have realized that the nowarn info does not have to be passed to the analyzer. Just references with appropriate nowarn attribute will be swallowed in the msbuild task

**Comment by dfederm** [comment]
> I have realized that the nowarn info does not have to be passed to the analyzer. Just references with appropriate nowarn attribute will be swallowed in the msbuild task

Yea that's what I was thinking. Just filter it on the "Declared" side.

**Comment by stan-sz** [comment]
> > I have realized that the nowarn info does not have to be passed to the analyzer. Just references with appropriate nowarn attribute will be swallowed in the msbuild task
> 
> Yea that's what I was thinking. Just filter it on the "Declared" side.

OTOH the current implementation would allow to report useless suppressions.

**Comment by erikmav** [comment]
Verified this allow me to fix the repo affected by #61 .

**Comment by erikmav** [comment]
Verified this allowed me to fix RT0002 issues in an affected repo. Found a NullRefException, sending PR for it.

**Pull request merged**